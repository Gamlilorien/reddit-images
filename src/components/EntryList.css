.entry-row {
    margin-bottom: 2em;
  }