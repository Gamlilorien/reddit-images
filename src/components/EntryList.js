import React from 'react';
import './EntryList.css';
import IconGroup from './IconGroup';
import Thumbnail from './Thumbnail';

const EntryList = (props) => {
    //We have to wrap our return statement in a div otherwise it will throw an error!
    return <div>
        {
            props.entries.map(entry => 
                <div className="row entry-row" key={entry.link}>
                    <div className="col-md-auto">
                        <Thumbnail entry={entry}/>
                    </div>
                    <div className="col">
                        <div className="row">
                            <h3>{entry.title}</h3>
                        </div>
                        <IconGroup entry={entry} />
                    </div>
                </div>
            )
        }
        </div>
};

export default EntryList;