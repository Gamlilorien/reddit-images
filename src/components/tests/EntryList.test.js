import React from 'react';
import renderer from 'react-test-renderer';
import * as ReactDomClient from 'react-dom/client'
import EntryList from '../EntryList';

//sample entry format
let entry = [
    {
        title: "Here Ya Go",
        thumb: "https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg",
        image: "https://i.redd.it/ecy7638h8n391.jpg",
        link: "https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/",
        created: 12344
    },
    {
        title: "Test entry #2",
        thumb: "https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg",
        image: "https://i.redd.it/ecy7638h8n391.jpg",
        link: "https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/",
        created: 155234
    }
]
const div = document.createElement("div");
const root = ReactDomClient.createRoot(div);

//afterEach(cleanup);

//make sure that the element gets created regarless of props
it("IconGroup element renders", () => {
    root.render(<EntryList />)
})


it("IconGroup matches test snapshot", () => {
    const tree = renderer.create(<EntryList entries={entry} />)
    expect(tree).toMatchSnapshot();
})