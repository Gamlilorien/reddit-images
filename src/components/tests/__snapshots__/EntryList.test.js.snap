// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconGroup matches test snapshot 1`] = `
<div>
  <div
    className="row"
    value="post"
  >
    <div
      className="sort-col col"
    >
      <div
        className="float-end"
      >
        <span>
          Sort by: 
        </span>
        <span
          className="active"
          onClick={[Function]}
          value="post"
        >
          Post |
        </span>
        <span
          className="not-active"
          onClick={[Function]}
          value="ascending"
        >
           A-z |
        </span>
        <span
          className="not-active"
          onClick={[Function]}
          value="descending"
        >
           Z-a
        </span>
      </div>
    </div>
  </div>
  <div
    className="row entry-row"
  >
    <div
      className="col-md-auto"
    >
      <a
        alt="open full size image in a new tab"
        href="https://i.redd.it/ecy7638h8n391.jpg"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Test entry #2"
          className="thumbnail"
          rel="noreferrer"
          src="https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg"
          target="_blank"
        />
      </a>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <h3>
          Test entry #2
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <img
            alt="click to favorite"
            className="blue-icon"
            onClick={[Function]}
            src="bookmark-regular.svg"
          />
          <span>
             | 
          </span>
          <a
            alt="Test entry #2"
            href="https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="view post"
              className="blue-icon"
              src="reddit-square-brands.svg"
            />
          </a>
          <span>
             | 
          </span>
          <a
            alt="open full size image in a new tab"
            href="https://i.redd.it/ecy7638h8n391.jpg"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="open full size"
              className="blue-icon"
              src="image-solid.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row entry-row"
  >
    <div
      className="col-md-auto"
    >
      <a
        alt="open full size image in a new tab"
        href="https://i.redd.it/ecy7638h8n391.jpg"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Here Ya Go"
          className="thumbnail"
          rel="noreferrer"
          src="https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg"
          target="_blank"
        />
      </a>
    </div>
    <div
      className="col"
    >
      <div
        className="row"
      >
        <h3>
          Here Ya Go
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col"
        >
          <img
            alt="click to favorite"
            className="blue-icon"
            onClick={[Function]}
            src="bookmark-regular.svg"
          />
          <span>
             | 
          </span>
          <a
            alt="Here Ya Go"
            href="https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="view post"
              className="blue-icon"
              src="reddit-square-brands.svg"
            />
          </a>
          <span>
             | 
          </span>
          <a
            alt="open full size image in a new tab"
            href="https://i.redd.it/ecy7638h8n391.jpg"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="open full size"
              className="blue-icon"
              src="image-solid.svg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
